### PR TITLE
Update snapcraft.yaml files for arm binaries

### LINKED ISSUE #55: error: snap "powershell" is not available on stable for this architecture (arm64)
[Install PowerShell on Linux](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-linux) does not mention Ubuntu 19.10.

I tried the 19.04 instructions instead, but it fails saying there is no PowerShell snap for arm64.

I'm running Ubuntu 19.10 Server on a Raspberry Pi 4 (so, ARM64).

# Steps to reproduce

```none
ubuntu@ubuntu:~$ sudo snap install powershell --classic
error: snap "powershell" is not available on stable for this architecture (arm64) but exists on other architectures (amd64).
```
# Expected behavior

Expect there to be a PowerShell snap for ARM64. 

# Actual behavior

```none
error: snap "powershell" is not available on stable for this architecture (arm64) but exists on other architectures (amd64).
```

# Environment data

```none
ubuntu@ubuntu:~$ hostnamectl
   Static hostname: ubuntu
         Icon name: computer
        Machine ID: c2e737ec4d1d40f8b7e6c381f9453a62
           Boot ID: 3f28773124a74a11b72e69a815d6c07d
  Operating System: Ubuntu 19.10
            Kernel: Linux 5.3.0-1014-raspi2
      Architecture: arm64
```

### LINKED ISSUE #73: Add arm64 snap to the snap store
Powershell is available for arm64 platforms, the snap should reflect this.

### PR DESCRIPTION
**Pull request by lucyllewy**
Add arm32 and arm64 binaries of PowerShell if the build is executed on an appropriate host.

To build an arm32 or arm64 version of the snap it will be necessary to modify the build-bots to execute the job on arm32 and arm64 hosts in addition to the current amd64. I believe it will also require additional `Dockerfile` definitions (at `./tools/releaseBuild/Images`) to create an appropriate `snapcraft` container for each new architecture.

Fixes #55 
Fixes #73

Signed-off-by: Daniel Llewellyn <daniel@bowlhat.net>

### CODE REVIEW

**Comment by lucyllewy** [comment]
It looks like the failure of CI is from unrelated changes upstream in Snapcraft: see https://github.com/snapcore/snapcraft/pull/2883

As we're building our own Docker images then the changes will need to be massaged into our Dockerfiles to add `core18` in addition to `core` to the Docker images.

**Comment by lucyllewy** [comment]
CI fixed. (Sorry it took me so long to circle back and resync against main)

**Comment by lucyllewy** [comment]
cc: @TravisEz13

File: stable/snap/snapcraft.yaml [hunk 1]
```diff
@@ -63,7 +63,14 @@ parts:
         echo $version > $file
       fi
       echo "getting powershell $version"
-      curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-x64.tar.gz
+      case "$SNAPCRAFT_ARCH_TRIPLET" in
+        aarch64-linux-gnu)
+          curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-arm64.tar.gz ;;
+        arm-linux-gnueabihf)
+          curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-arm32.tar.gz ;;
+        *)
```
Comment (TravisEz13) [inline stable/snap/snapcraft.yaml]: Can we be more explicit here?

File: preview/snap/snapcraft.yaml [hunk 2]
```diff
@@ -61,7 +61,14 @@ parts:
         echo $version > $file
       fi
       echo "getting powershell $version"
-      curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-x64.tar.gz
+      case "$SNAPCRAFT_ARCH_TRIPLET" in
+        aarch64-linux-gnu)
+          curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-arm64.tar.gz ;;
+        arm-linux-gnueabihf)
+          curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-arm32.tar.gz ;;
+        *)
```
Comment (TravisEz13) [inline preview/snap/snapcraft.yaml]: Can we be more explicit here?

File: stable/snap/snapcraft.yaml [hunk 3]
```diff
@@ -63,7 +63,14 @@ parts:
         echo $version > $file
       fi
       echo "getting powershell $version"
-      curl -L -o powershell.tar.gz https://github.com/PowerShell/PowerShell/releases/download/v$version/powershell-$version-linux-x64.tar.gz
+      case "$SNAPCRAFT_ARCH_TRIPLET" in
```
Comment (TravisEz13) [inline stable/snap/snapcraft.yaml]: Perhaps we only make these changes to preview at first to minimize risk.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
We need to test this, but we don't currently have ARM.  Do you know a public CI system that can run ARM?

File: preview/snap/snapcraft.yaml (hunk 2 above)
Comment (lucyllewy) [reply preview/snap/snapcraft.yaml]: we can add an explicit `x86_64-linux-gnu` block and then fail if it doesn't match any of the three options..

File: stable/snap/snapcraft.yaml (hunk 3 above)
Comment (lucyllewy) [reply stable/snap/snapcraft.yaml]: ok, I'll drop the stable snapcraft for the time being

**Comment by lucyllewy** [comment]
unfortunately I don't know about any available CI systems that have an ARM runner. I've been on the lookout for other snap projects, too..

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
Thanks for the updates.

**Comment by TravisEz13** [comment]
I don't have a build environment for this, but I have been promised one "soon".

**Pull request merged**